doc/build/changelog/unreleased_20/8547.rst

```diff
@@ -0,0 +1,9 @@
+.. change::
+    :tags: bug, engine
+    :tickets: 8547
+
+    For improved security, :class:`engine.URL` gets its ``__str__`` method
+    removed to avoid accidental password leakage. ``str(engine.URL())`` will
+    now fallback to :method:`engine.URL.__repr__` which censors the password.
+    :method:`engine.URL.render_as_string(hide_password=False)` is still
+    available for cases when clear text password is necessary.
```

Comment:
fell free to add "Pull request courtesy of Yassen Damyanov" if you want

Reply:
I would rather not if that's okay, thanks @CaselIT.

Reply:
as you prefer

Reply:
Just to confirm @ya55en , we do normally credit authors for PRs in the changelog.  I am finalizing your PR now, I can absolutely omit that here if you prefer, I will proceed that way.

Reply:
I would prefer to omit any credit, if this is not inappropriate. It was actually @CaselIT 's idea to remove `URL.__str__` entirely, so really no need for my name to pop up in the Changelog IMHO. Thanks @zzzeek.

Reply:
I'm still going to thank "our contributors" in the abstract.   You won't be named however people can see your email in the git commit log which of course lights up on github and in your profile and all that.   Do you want me to scrub that also ?   (note that of course your PR here is public and that's already done).

Reply:
> going to thank "our contributors" in the abstract

Sure, this is fine 👍

> note that of course your PR here is public

Yeah, of course, I am not trying to hide, you know... 🙂 Thanks.